beacon-chain/state/state-native/getters_validator.go

```diff
@@ -257,23 +332,22 @@ func (b *BeaconState) InactivityScores() ([]uint64, error) {
 		return nil, errNotSupported("InactivityScores", b.version)
 	}
 
-	if b.inactivityScores == nil {
-		return nil, nil
-	}
-
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 
 	return b.inactivityScoresVal(), nil
 }
 
-// inactivityScoresVal of validators participating in consensus on the beacon chain.
-// This assumes that a lock is already held on BeaconState.
 func (b *BeaconState) inactivityScoresVal() []uint64 {
+	if features.Get().EnableExperimentalState {
+		if b.inactivityScoresMultiValue == nil {
+			return nil
+		}
+		return b.inactivityScoresMultiValue.Value(b)
```

Comment:
Does value copy the whole list ?

Reply:
yes, but currently we also copy by using `copy`